### PR TITLE
Fixes #2143 - Support listing branch/pr per commit

### LINKED ISSUE #2143: preview support for listing branches or pull requests for a commit
- https://developer.github.com/v3/repos/commits/#list-branches-for-head-commit
 - https://developer.github.com/v3/repos/commits/#list-pull-requests-associated-with-commit

### PR DESCRIPTION
**Pull request by zacdirect**

### CODE REVIEW

**Comment by zacdirect** [comment]
I'm happy to make any adjustments.  The one major thing I was on the fence about was creating a new model for CommitPullRequests.  It could easily be returned as the existing PullRequest model, but in sampling some responses from GitHub I was never getting information back for the properties I removed.  If their absence is possibly explained by a different preview header or license level then it would make sense to return them as the existing PullRequest model.  I was unable to find any documentation explaining the difference so I opted for a new model to make it clear those PullRequest properties will never be there for these calls.

**Comment by zacdirect** [comment]
Is this repo still active?

**Comment by timrogers** [comment]
> Is this repo still active?

Yes!

**Comment by matt-richardson** [comment]
Can we re-open this PR and look to get it merged?

**Comment by kfcampbell** [comment]
I've reopened the PR and fixed the merge conflict.

**Review by nickfloyd — APPROVED** [review]
Everything looks good, thank you for these contributions!  ❤️ .  I just left one comment about the preview header.  Not a show stopper, but it's essentially dead code and unnecessary, given REST API previews were deprecated last year.  We can remove the code in this PR or do a follow-up if needed.

File: Octokit/Helpers/AcceptHeaders.cs [hunk 1]
```diff
@@ -13,5 +13,7 @@ public static class AcceptHeaders
         /// </summary>
         /// <remarks>https://developer.github.com/v3/repos/contents/#custom-media-types</remarks>
         public const string RawContentMediaType = "application/vnd.github.v3.raw";
+
+        public const string ListBranchOrPullForCommitPreview = "application/vnd.github.groot-preview+json";
```
[the remark this answers is not in the document]
Comment (kfcampbell) [reply Octokit/Helpers/AcceptHeaders.cs]: I've removed the offending header!

**Comment by matt-richardson** [comment]
Hi @nickfloyd - is there anything stopping this one from being merged?

**Comment by nickfloyd** [comment]
> Hi @nickfloyd - is there anything stopping this one from being merged?

I think it's g2g.  Let me give it a once over and get it merged in. Apologies for the delay.

**Review by nickfloyd — APPROVED** [review]
LGTM ❤️

**Pull request merged**